### ISSUE TITLE
Fix scimPatch type signature to include generics.

### ISSUE THREAD

**Issue by markwongsk** [thomaspoignant/scim-patch#41]
**Is your feature request related to a problem? Please describe.**
The type signature of `scimPatch` requires unnecessary type casts on the consumer end.

**Describe the solution you'd like**
Instead of 
```
scimPatch(scimResource: ScimResource, patchOperations: Array<ScimPatchOperation>): ScimResource
``` 
the signature
```
scimPatch<T extends ScimResource>(scimResource: T, patchOperations: Array<ScimPatchOperation>): T
```
Allows the consumer to pass in their own ScimResource type.

**Describe alternatives you've considered**
Manually typecasting the result back eg
```
scimPatch(resource, ops) as MyScimResourceType;
```

**Additional context**
Here's a proposed PR: https://github.com/markwongsk/scim-patch/pull/1 which I can open here if this change sounds reasonable.

**Comment by thomaspoignant** [comment]
I like the idea, I need to check a bit more in detail but it seems to work well from what I see.
Can you open the PR so we will be able to talk about the code?

**Comment by thomaspoignant** [comment]
@markwongsk thanks for your PR this is awesome, I have merged it.
It will be available in the version `v0.3.0` of `scim-patch`.

For your information, I've also updated the test to use the generics and remove the cast we were using.
See #47 for more details.
